app/src/main/java/com/guillermorosales/smarthome/MainViewModel.kt

```diff
@@ -0,0 +1,34 @@
+package com.guillermorosales.smarthome
+
+import android.arch.lifecycle.MutableLiveData
+import android.arch.lifecycle.ViewModel
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.FirebaseDatabase
+import com.google.firebase.database.ValueEventListener
+
+class MainViewModel : ViewModel() {
+    private lateinit var lightState: MutableLiveData<Boolean>
+
+    fun getLightState(): MutableLiveData<Boolean> {
+        if (!::lightState.isInitialized) {
+            lightState = MutableLiveData()
+            FirebaseDatabase.getInstance().reference.child("lights").child("on").addValueEventListener(object : ValueEventListener {
```

Comment:
I couldn't find where you are initializing firebase but looks like we could use some DI here 😄

Reply:
oh yeah,  firebase is being initialized automatically with apply plugin: 'com.google.gms.google-services' and the json file but yeah we could inject the DB reference on the view model.